lib/type.ts

```diff
@@ -138,10 +138,10 @@ export default abstract class Type implements Hashable<Type> {
         return type;
     }
 
-    static isAssignable(type : Type, assignableTo : Type|string, typeScope : TypeScope = {}, entitySubTypeMap : EntitySubTypeMap = {}) : boolean {
+    static isAssignable(type : Type, assignableTo : Type|string, typeScope : TypeScope = {}, entitySubTypeMap : EntitySubTypeMap = {}, relax = false) : boolean {
```

Comment:
Ugh relax is back. What are the semantics of relax now?

Reply:
> Ugh relax is back. What are the semantics of relax now?

when `relax` is true, `a` is assignable to `b` if they share common subtypes. I added `relax` to avoid breaking existing code just in case... 
i agree it's confusing... do you think we can make this the standard for everything?